tokio/src/sync/set_once.rs

```diff
@@ -0,0 +1,402 @@
+use super::Notify;
+
+use crate::loom::sync::{atomic::AtomicBool, Mutex};
+use crate::{loom::cell::UnsafeCell, pin};
+
+use std::error::Error;
+use std::fmt;
+use std::mem::MaybeUninit;
+use std::ops::Drop;
+use std::ptr;
+use std::sync::atomic::Ordering;
+
+// This file contains an implementation of an SetOnce. The value of SetOnce
+// can only be modified once during initialization.
+//
+//  1. When `value_set` is false, the `value` is not initialized and wait()
+//      future will keep on waiting.
+//  2. When `value_set` is true, the wait() future completes, get() will return
+//      Some(&T)
+//
+// The value cannot be changed after set() is called. Subsequent calls to set()
+// will return a `SetOnceError`.
+
+/// A thread-safe cell that can be written to only once. A `SetOnce`
+/// is inspired from python's [`asyncio.Event`] type. It can be used to wait
+/// until the value of the `SetOnce` is set like a "Event" mechanism.
```

Comment:
A newline here looks nicer in the docs.
```suggestion
/// A thread-safe cell that can be written to only once. 
///
/// A `SetOnce` is inspired from python's [`asyncio.Event`] type. It can be
/// used to wait until the value of the `SetOnce` is set like a "Event"
/// mechanism.
```